test/wowman/config_test.clj

```diff
@@ -26,45 +27,61 @@
-      (is (= expected (config/configure file-opts cli-opts)))))
+      (is (= expected (config/merge-config file-opts cli-opts)))))
 
-  (testing "cli overrides file overrides"
+  (testing "cli options override file options"
     (let [cli-opts {:debug? true}
           file-opts {:debug? false}
           expected (assoc config/default-cfg :debug? true)]
-      (is (= expected (config/configure file-opts cli-opts))))))
+      (is (= expected (config/merge-config file-opts cli-opts))))))
 
-(deftest invalid-config
+(deftest invalid-addon-dirs-in-cfg
   (testing "missing directories don't nuke entire config"
-    (let [user-file-config {:addon-dir-list [{:addon-dir (str fs/*cwd*) ;; exists
-                                              :game-track "classic"}
-                                             {:addon-dir "/does/not/exist"
-                                              :game-track "retail"}]
-                            :debug? false
-                            :selected-catalog :short}
+    (let [cli-opts {}
+          file-opts {:addon-dir-list [{:addon-dir (str fs/*cwd*) ;; exists
+                                       :game-track "classic"}
+                                      {:addon-dir "/does/not/exist"
+                                       :game-track "retail"}]}
+          expected (assoc config/default-cfg
+                          :addon-dir-list [{:addon-dir (str fs/*cwd*)
+                                            :game-track "classic"}])]
+      (is (= expected (config/merge-config file-opts cli-opts))))))
 
-          expected-config {:addon-dir-list [{:addon-dir (str fs/*cwd*)
-                                             :game-track "classic"}]
-                           :debug? false
-                           :selected-catalog :short}
-          cli-opts {}]
-      (is (= expected-config (config/configure user-file-config cli-opts))))))
+(deftest load-settings-from-file
+  (testing "a standard config file circa 0.10 is loaded and parsed as expected"
+    (let [fixture (fixture-path "user-config-0.10.json")
+          cli-opts {}
+          expected {:selected-catalog :full,
+                    :debug? true,
+                    :addon-dir-list [{:addon-dir "/home/torkus/bar", :game-track "retail"}
+                                     {:addon-dir "/home/torkus/foo", :game-track "classic"}]}]
+      (is (= expected (config/load-settings-file fixture)))))
+
+  (testing "a standard config file circa 0.11 is loaded and parsed as expected"
+    (let [fixture (fixture-path "user-config-0.11.json")
+          cli-opts {}
+          expected {:selected-catalog :full,
+                    :debug? true,
+                    :gui-theme :light,
+                    :addon-dir-list [{:addon-dir "/home/torkus/bar", :game-track "retail"}
+                                     {:addon-dir "/home/torkus/foo", :game-track "classic"}]}]
+      (is (= expected (config/load-settings-file fixture))))))
```

Comment:
this isn't too useful. I want to see old configuration passed through current logic